### PR TITLE
react: replace useOperation with useDeferredTask

### PR DESCRIPTION
**Pull request by krzysu**

### CODE REVIEW

File: packages/react/src/publication/useHidePublication.ts [hunk 1]
```diff
@@ -2,46 +2,44 @@ import { AnyPublication } from '@lens-protocol/api-bindings';
...
  * function HidePublication({ publication }: { publication: AnyPublication }) {
- *  const { execute: hide, isPending } = useHidePublication({ publication });
+ *   const { execute: hide, loading } = useHidePublication({ publication });
  *
- *  return (
- *    <button onClick={hide} disabled={isPending}>
- *      Hide
- *    </button>
- * );
- *```
```
Comment (krzysu) [inline packages/react/src/publication/useHidePublication.ts:35]: as you notice, this example snippet is broken and there is no way for us to auto-detect it, so the more content we write inside comment blocks the more content we don't really control

Comment (cesarenaldi) [reply packages/react/src/publication/useHidePublication.ts:35]: This is true regardless of where we write examples. Here you spotted it 👍

File: packages/react/src/publication/useHidePublication.ts [hunk 2]
```diff
@@ -2,46 +2,44 @@ import { AnyPublication } from '@lens-protocol/api-bindings';
...
+ *   return (
+ *     <button onClick={hide} disabled={loading}>
+ *       Hide
+ *     </button>
+ *   );
+ * }
+ * ```
+ *
+ * @category Publications
+ * @group Hooks
```
Comment (cesarenaldi) [inline packages/react/src/publication/useHidePublication.ts:33]:
```suggestion
 * @group Hooks
 * @param args - {@link UseHidePublicationArgs}
```

File: packages/react/src/publication/useReportPublication.ts [hunk 3]
```diff
@@ -91,29 +84,30 @@ export type ReportPublicationOperation = Operation<void, never, [ReportPublicati
...
+ *       <button type="submit" disabled={loading}>
  *         Report
  *       </button>
  *     </form>
  *   );
  * }
  * ```
+ *
+ * @category Publications
+ * @group Hooks
```
Comment (cesarenaldi) [inline packages/react/src/publication/useReportPublication.ts:100]:
```suggestion
 * @group Hooks
 * @param args - {@link UseReportPublicationArgs}
```

**Pull request merged**